### ISSUE TITLE
The three readers of one neighbour's whole setting look under one namespace only, so a copy under another namespace writes unseen

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#715]
### What minimal example or steps are needed to reproduce the bug?

`neighbourSetting` — the reader of one neighbour's setting with its secondary options, beside `neighbourSettings` — looks the neighbour up as `addNamespace(rule.name, syntax.namespace)`, the asking rule's namespace alone. Every namespace reads a plain CSS file, so a neighbour listed under another one writes the same file and is not seen. #710 widened `neighbourSettings`; its three readers remain, and each needs its own answer to "which of two copies":

- `writesBlockAfter` (`block-opening-brace-newline-after` over a block of comments) should ask every copy that writes;
- `closedBySemicolon` (the two `declaration-colon-*-after` rules asking `declaration-block-trailing-semicolon`) should ask the copy that runs last, and read `fixDisabledOnLine` under its name;
- `no-multiple-whitespaces` asking `named-grid-areas-alignment` whether it lays a table out should ask whether any copy does.

```css
/* input */
a { /* c */ }
```

```shell
# `block-opening-brace-newline-after: "always"`, `block-closing-brace-newline-before: "never-multi-line"`, both under the core
--fix leaves `a { /* c */ }`; a check reports the opening brace rule

# the same with the second rule as `scss/block-closing-brace-newline-before`
--fix leaves `a { /* c */}`; a check reports the opening brace rule
```

```css
/* input */
a { color: ; }
```

```shell
# `declaration-colon-space-after: "never"` then `declaration-block-trailing-semicolon: "never"`, both under the core
--fix leaves `a { color:}` in one run

# the same with `scss/declaration-block-trailing-semicolon`
the first run leaves `a { color: }`, the second `a { color:}`
```

```css
/* input */
a {
	grid-template:
		[a] "x  x" 1fr
		[bbb] "y y" auto  / 1fr;
}
```

```shell
# `scss/named-grid-areas-alignment: [true, { alignColumns: true }]` then `no-multiple-whitespaces: true`
--fix leaves `[a] "x x" 1fr`, and a check reports `named-grid-areas-alignment`; under the core's name both rules rest on `[a]   "x x" 1fr`
```

The same reading grows a file without end where one rule is listed under both names with opposite options:

```css
/* input */
a { color:; }
```

```shell
# `declaration-block-trailing-semicolon: "always"`, `scss/declaration-block-trailing-semicolon: "never"`, `declaration-colon-space-after: "always"`
--fix leaves `a { color:  }`, then `a { color:   }`, then `a { color:    }`, a space more on every run, on `main` and on the branch of #710 alike
```

Measured on the branch of #710 by a probe splitting pairs of neighbouring rules between the core and `scss/`: of 8 920 configurations, 52 split runs over `writesBlockAfter` and 12 over `closedBySemicolon` still part from the same pairs under the core.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/block-opening-brace-newline-after": "always",
    "@stylistic/scss/block-closing-brace-newline-before": "never-multi-line"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

A reader of a neighbour's setting finds the neighbour under every namespace that reads the root, as `neighbourSettings` does since #710.

### What actually happened?

It finds only the copy under the asking rule's namespace, and the copy under the other one writes as if nothing had asked.

### Do you have a proposal to fix the bug?

Return every copy from `neighbourSetting`, with its name, through the same matching `neighbourSettings` uses, and let each of the three readers decide over the list. `writesRunBehindBrace` of #698 reads `ignoreAtRules` through it as well.

Found while fixing #710.
